# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../dietlibc/patches/pkg_pciutils.patch
# Copyright (C) 2006 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- ./lib/types.h.orig	2006-06-21 14:44:57.000000000 +0200
+++ ./lib/types.h	2006-06-21 14:56:35.000000000 +0200
@@ -15,9 +15,9 @@
 typedef unsigned __int16 u16;
 typedef unsigned __int32 u32;
 #else
-typedef u_int8_t u8;
-typedef u_int16_t u16;
-typedef u_int32_t u32;
+typedef uint8_t u8;
+typedef uint16_t u16;
+typedef uint32_t u32;
 #endif
 
 #ifdef PCI_HAVE_64BIT_ADDRESS
--- ./lib/pread.h.orig	2006-06-21 14:59:19.000000000 +0200
+++ ./lib/pread.h	2006-06-21 14:59:41.000000000 +0200
@@ -12,7 +12,7 @@
  *  don't define it.
  */
 
-#if defined(__GLIBC__) && __GLIBC__ == 2 && __GLIBC_MINOR__ > 0
+#if (defined(__GLIBC__) && __GLIBC__ == 2 && __GLIBC_MINOR__ > 0 || defined(__dietlibc__))
 /* glibc 2.1 or newer -> pread/pwrite supported automatically */
 
 #elif defined(i386) && defined(__GLIBC__)
